include/ibm/management_console_rest.hpp

```diff
@@ -32,58 +37,94 @@ constexpr const char* methodNotAllowedMsg = "Method Not Allowed";
-    500000;   // Allow save area file size upto 500KB
+    25000000; // Allow save area file size upto 25MB
 constexpr size_t maxBroadcastMsgSize =
     1000;     // Allow Broadcast message size upto 1KB
 
-inline void handleFilePut(const crow::Request& req,
-                          const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
-                          const std::string& fileID)
+inline bool isValidConfigFileName(const std::string& fileName,
+                                  nlohmann::json& resp)
 {
-    std::error_code ec;
-    // Check the content-type of the request
-    boost::beast::string_view contentType = req.getHeaderValue("content-type");
-    if (!bmcweb::asciiIEquals(contentType, "application/octet-stream"))
+    if (fileName.empty())
     {
-        asyncResp->res.result(boost::beast::http::status::not_acceptable);
-        asyncResp->res.jsonValue["Description"] = contentNotAcceptableMsg;
-        return;
+        BMCWEB_LOG_ERROR("Empty filename");
+        resp["Description"] = "Empty file path in the url";
+        return false;
     }
-    BMCWEB_LOG_DEBUG(
-        "File upload in application/octet-stream format. Continue..");
 
-    BMCWEB_LOG_DEBUG(
-        "handleIbmPut: Request to create/update the save-area file");
-    std::string_view path =
-        "/var/lib/bmcweb/ibm-management-console/configfiles";
-    if (!crow::ibm_utils::createDirectory(path))
+    // ConfigFile name is allowed to take upper and lowercase letters,
+    // numbers and hyphen
+    std::size_t found = fileName.find_first_not_of(
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-");
+    if (found != std::string::npos)
     {
-        asyncResp->res.result(boost::beast::http::status::not_found);
-        asyncResp->res.jsonValue["Description"] = resourceNotFoundMsg;
-        return;
+        BMCWEB_LOG_ERROR("Unsupported character in filename: {}", fileName);
+        resp["Description"] = "Unsupported character in filename";
+        return false;
     }
 
+    // Check the filename length
+    if (fileName.length() > 20)
+    {
+        BMCWEB_LOG_ERROR("Name must be maximum 20 characters. "
+                         "Input filename length is: {}",
+                         fileName.length());
+        resp["Description"] = "Filename must be maximum 20 characters";
+        return false;
+    }
+
+    return true;
+}
+
+inline bool saveConfigFile(const std::string& data, const std::string& fileID,
+                           const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    std::error_code ec;
+    // Get the file size getting uploaded
+    BMCWEB_LOG_DEBUG("data length: {}", data.length());
+    BMCWEB_LOG_DEBUG("fileID: {} ", fileID);
+
+    if (data.length() < minSaveareaFileSize)
+    {
+        asyncResp->res.result(boost::beast::http::status::bad_request);
+        asyncResp->res.jsonValue["Description"] =
+            "File size is less than minimum allowed size[100B]";
```

Comment:
It looks like `100B` is defined as minSaveareaFileSize.
Should we write this like?

```
        asyncResp->res.jsonValue["Description"] = std::format(
            "File size is less than minimum allowed size[{}]", minSaveareaFileSize);
```